### PR TITLE
EN-2205 EN-2224 Improved error handling on PAM widgets config

### PR DESCRIPTION
**Pull request by zonia3000**

### CODE REVIEW

**Comment by entando-service** [comment]
All is Well

**Comment by zonia3000** [comment]
The classes `BpmFormWidgetAction` and `BpmBpmCaseInstanceSelectorWidgetAction` have a lots of methods in common however they inherit from 2 different classes, so, to avoid code duplication both on these classes and their tests I created an interface `BpmSourceAndProcessSelector` and a class `BpmSourceAndProcessSelectorHelper` that wraps the logic for loading the knowledge sources and processes dropdown values, handling also the error conditions (config not found, PAM instance not reachable, PAM instance disabled).
In the case of disabled instances I also added the text "(Disabled)" near the name of the instance in the dropdown, so that for the user it is clear that instance is not active.

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmDatatableWidgetAction.java [hunk 1]
```diff
@@ -133,35 +121,35 @@ protected BpmWidgetInfo storeWidgetInfo(final Widget widget) {
...
     @Override
     public String chooseForm() {
+        String response = super.chooseForm();
         try {
             this.loadFieldIntoDatatableFromBpm();
-        } catch (Exception t) {
-            _logger.error("Error in chooseForm", t);
-            return FAILURE;
+        } catch (ApsSystemException t) {
+            logger.error("Error in chooseForm", t);
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmDatatableWidgetAction.java]: Should this still set response = FAILURE? Seems like an important logical change

Comment (zonia3000) [reply plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmDatatableWidgetAction.java]: Fixed

**Comment by entando-service** [comment]
All is Well

**Pull request merged**